contracts/GovHub.template

```diff
@@ -214,11 +215,11 @@ contract GovHub is System, IParamSubscriber {
       4. the sender's vote is executed by incrementing the proposal's for or against counter
       5. the sender is marked as having voted
 */
-  function castVote(uint256 proposalId, bool support) 
+  function castVote(uint256 proposalId, bool support) //@openissue
```

Comment:
I'm not sure the logic of this flow is consistent.
at proposal's creation we set the totalVotes field to the current number of members clearly indicating  that voting is limited to pre-proposal members (else no need for this value). still, castVote makes no such checks and allows any member, regardless of entry date, to vote.

I'm not even saying that the latter is bad, it's ok to start simple, but if that is the  case let's throw away totalVotes and rely on the current number of members

Reply:
my point is this:
a. we stabilize totalVotes at proposal creation indicating that we only care about members that existed at the 
 moment of proposal creation
b. at voting period we allow _all_ members to vote - even those who joined after proposal creation

if we want only members prior to props creation to vote - we need to enforce it.
if we want all members to vote - why use the old totalVotes rather than the current?

Reply:
I think the approach with the least downsides is to store the member list on each proposal on proposal creation and restrict voting to those members.

Another potential solution is preventing new members from joining while proposals are open (other than the `addMember` proposal). But then instead of iterating over previous proposals to check their status, we'd probably need to add a new state variable `activeProposalCount` that tracks the number of open proposals so we can check against it. Imo it starts to get unreasonably complex, which if why I prefer the first option despite the additional storage cost for the proposal creator.